Skyline.DataMiner.CICD.Tools.DataminerDeploy.Lib/DataMinerService/SLNetDataMinerService.cs

```diff
@@ -0,0 +1,438 @@
+                    using (System.IO.StreamReader reader = new System.IO.StreamReader(stream))
+                    {
+                        sb.AppendLine(reader.ReadToEnd());
+                    }
+                }
+
+                int build;
+
+                var appInfoXml = XDocument.Parse(appInfoSb.ToString());
+                string buildStr = appInfoXml?.Root?.Element("Build")?.Value;
+                if (buildStr != null && Int32.TryParse(buildStr, out int tryParsedAppBuild))
+                {
+                    build = tryParsedAppBuild;
+                }
+                else
+                {
+                    build = 0;
+                }
+
+                string name = appInfoXml?.Root?.Element("Name")?.Value;
+                string version = appInfoXml?.Root?.Element("Version")?.Value;
+
+                return (name, version, build);
+            }
+        }
+
+        private void OnResolveConnection(object sender, ResolveConnectionArgs e)
+        {
+            if (e.DestinationIPAddress.Equals("localhost") || e.DestinationIPAddress.Equals("127.0.0.1"))
+            {
+                e.Connection = null;
+                return;
+            }
+
+            e.Connection = slnet.Connection;
+        }
+
+        private void OnUpdate(object sender, UpgradeWatcherLogEventArgs e)
+        {
+            logger.LogDebug(e.ProgressInfo.Message);
+        }
+
+        private void VerifyMinimumDataMinerVersion()
+        {
+            // This has a minimum DataMiner requirement:
+            // https://dms-it-dev.skyline.local/apigateway/api/version
+            // need
+            //	APIGateway versie >"0.0.1.52414"
+            //	SLNetUpgradeService > "10.3.2243.10098"
+            //	SLNetUpgradeService > 10.3.2317.354
+
+            // Should verify the remote agent first, then make a new clean error message otherwise u get an unclear exception from slnet.
+            // minimum dataminer version: https://intranet.skyline.be/DataMiner/Lists/Release%20Notes/DispForm2.aspx?ID=36023
+            //		Main Release Version
+            //		10.3.0[CU3]
+            //		Feature Release Version
+            //		10.3.6
+            //		Internal Build ID
+            //		10.3.0.0 - 12948
+            // Note you need: .NET Hosting Bundle before upgrading agent, otherwise it won't be able to update the APIGateway
+
+            // Let's avoid needing to make a different connection. Let's re-use the SLNetCommunication here.
+            // Assuming there might be corner cases where retrieve the version won't work.
+            // At that case we'll best effort try the upgrade and let any exceptions bubble up from the core if they would happen.
+            // The goal of the validation is to stop the code (the core software will do that), it's to make a clear error message to the user if we detect an invalid version.
+
+            string minimumRequired = "10.3.0.0";
```

Comment:
As it's a specific CU, then maybe add the build number? DataMinerVersion class from Skyline.DataMiner.CICD.Common can be used.

Reply:
Using the code from the DataMinerSystem library to retrieve with SLNET call the Agent Version. This one seems to be coming back without CU or Build info. Hence the best effort of checking with just the 4 versions.

Reply:
With the above comment, then it could be feasible to check build numbers.